src/bunit.core/Rendering/TestRenderer.cs

```diff
@@ -148,13 +156,28 @@ public void DisposeComponents()
 	protected override Task UpdateDisplayAsync(in RenderBatch renderBatch)
 	{
 		logger.LogNewRenderBatchReceived();
-
+		Dispatcher.AssertAccess();
 		var renderEvent = new RenderEvent(renderBatch, new RenderTreeFrameDictionary());
+		
+
+		if (syncContext is not null)
+		{
+			syncContext.Send(_ => UpdateRenderedComponents(renderEvent), null);
+		}
+		else
+		{
+			UpdateRenderedComponents(renderEvent);
+		}
```

Comment:
```suggestion
			UpdateRenderedComponents(renderEvent);
```

It's interesting that if we change the code to look like this, none of the tests otherwise fail when using UIFact and UITheory, stop doing so on my computer.